tests/test_search.py

```diff
@@ -1,60 +1,205 @@
+        self.assertEqual(len(search_response), 2)
+        self.assertEqual(len(search_response['results']), 1)
+        self.assertEqual(search_response['results'][0]['bundle_id'], bundle_id)
+        self.assertEqual(search_response['results'][0]['bundle_url'], bundle_url)
+
+    def test_search_returns_no_results_when_no_documents_indexed(self):
+        self.verify_search_results(smartseq2_paired_ends_query)
+
+    def test_search_returns_no_result_when_query_does_not_match_indexed_documents(self):
         query = \
             {
                 "query": {
-                    "bool": {
-                        "must": [{
-                            "match": {
-                                "files.sample_json.donor.species": "Homo sapiens"
-                            }
-                        }, {
-                            "match": {
-                                "files.assay_json.single_cell.method": "Fluidigm C1"
-                            }
-                        }, {
-                            "match": {
-                                "files.sample_json.ncbi_biosample": "SAMN04303778"
-                            }
-                        }]
+                    "match": {
+                        "files.sample_json.donor.species": "xxx"
                     }
                 }
             }
-        self.assertPostResponse(
-            '/v1/search',
-            json_request_body=(query),
-            expected_code=requests.codes.ok)
+
+        self.populate_search_index(self.index_document, 1)
+        self.verify_search_results(query)
+
+    def test_search_returns_error_when_invalid_query_used(self):
+        # Some types of invalid queries are detected by Elasticsearch DSL
+        # and others by Elasticsearch itself, and the response codes differ.
+        invalid_query_data = [
+            (
+                {
+                    "query": {
+                        "mtch": {
+                            "SomethingInvalid": "xxx"
+                        }
+                    }
+                },
+                requests.codes.bad_request
+            ),
+            (
+                {
+                    "query": {
+                        "match": ["SomethingInvalid", "xxx"]
+                    }
+                },
+                requests.codes.internal_server_error
+            )
+        ]
+
+        self.populate_search_index(self.index_document, 1)
+        for query_data in invalid_query_data:
+            with self.subTest("Invalid Queries", bad_query=query_data):
+                self.assertPostResponse(
+                    "/v1/search",
+                    json_request_body=(query_data[0]),
+                    expected_code=query_data[1])
+
+    def test_search_returns_X_results_when_X_documents_match_query(self):
+        """The number of documents indexed is equal to the number of search results returned.
```

Comment:
Suggestion:

```
This test creates N identical documents.  It executes a search query that matches the document and verifies all the documents are returned.  N is varied across a variety of values.
```

Also:

1. What's the performance impact of this test?  How long do unit tests take now?
2. Do we need all the cases?

Reply:
1. This test takes 24s to run. All unit tests take about 7min
2. All of the test cases are needed especially as we continue to implement pagination. 0 and 1 for edge cases. 9, 10, 11 also edge cases since our original search implementation only returned 10 results max. 1000 is the max number of results currently return per call to scan. 5000 to test if all results are returned when multiple calls to scan are required.

Reply:
Awesome, it's good to know this does not substantially increase the runtime of tests. :)